skyportal/facility_apis/lt.py

```diff
@@ -0,0 +1,855 @@
+        response = client.service.handle_rtml(full_payload).replace(
+            'encoding="ISO-8859-1"', ''
+        )
+        response_rtml = etree.fromstring(response)
+        mode = response_rtml.get('mode')
+
+        if mode == 'confirm':
+            request.status = 'submitted'
+        else:
+            request.status = f'rejected: {mode}'
+
+        transaction = FacilityTransaction(
+            request=http.serialize_requests_request_xml(full_payload),
+            response=http.serialize_requests_response_xml(response),
+            followup_request=request,
+            initiator_id=request.last_modified_by_id,
+        )
+
+        DBSession().add(transaction)
+
+    _observation_types = ['r', 'gr', 'gri', 'griz', 'ugriz']
+    _exposure_types = ['1x120s', '2x150s']
+
+    form_json_schema = {
+        "type": "object",
+        "properties": {
+            "observation_type": {
+                "type": "string",
+                "enum": _observation_types,
+                "default": "r",
+            },
+            "exposure_type": {
+                "type": "string",
+                "enum": _exposure_types,
+                "default": '1x120s',
+            },
+            "priority": {"type": "string", "enum": ["1", "5"], "default": "1"},
+            "LT_proposalID": {"type": "string"},
+            "start_date": {
+                "type": "string",
+                "format": "date",
+                "default": datetime.utcnow().date().isoformat(),
+                "title": "Start Date (UT)",
+            },
+            "end_date": {
+                "type": "string",
+                "format": "date",
+                "title": "End Date (UT)",
+                "default": (datetime.utcnow().date() + timedelta(days=7)).isoformat(),
+            },
+            "maximum_airmass": {
+                "title": "Maximum Airmass (1-3)",
+                "type": "number",
+                "default": 2.0,
+            },
+            "maximum_seeing": {
+                "title": "Maximum Seeing [arcsec] (0-5)",
+                "type": "number",
+                "default": 1.2,
+            },
+            "sky_brightness": {
+                "title": "Maximum allowable Sky Brightness, Dark + X magnitudes [arcsec] (0-5)",
+                "type": "number",
+                "default": 2.0,
+            },
+            "photometric": {
+                "title": "Does this observation require photometric conditions?",
+                "type": "boolean",
+            },
+            "binning": {"type": "string", "enum": ["1x1", "2x2"], "default": "1x1"},
+        },
+        "required": [
+            "observation_type",
+            "exposure_type",
+            "priority",
+            "start_date",
+            "end_date",
+            "maximum_airmass",
+            "maximum_seeing",
+            "binning",
+        ],
+    }
+
+    ui_json_schema = {}
+
+
+class IOIAPI(LTAPI):
```

Comment:
IOO and IOI have many of the same methods and data, perhaps define a base class with the shared methods and attributes and subclass this as needed for each instrument